app/src/org/odk/collect/android/views/media/AudioButton.java

```diff
@@ -22,40 +21,61 @@
  * @author ctsims
  * @author carlhartung
  * @author amstone326
+ * @author Phillip Mates (pmates@dimagi.com)
  */
 public class AudioButton extends ImageButton implements OnClickListener {
-    private final static String t = "AudioButton";
+    private final static String TAG = AudioController.class.getSimpleName();
+
+    /**
+     * Audio to load when play button pressed.
+     */
     private String URI;
+
+    /**
+     * Audio playback state, used for correctly displaying the button and
+     * dispatching playback logic on button presses.
+     */
     private MediaState currentState;
-    private final AudioController controller;
-    private Object residingViewId;
 
+    /**
+     * The id of the ListAdapter view that contains this button
+     */
+    private ViewId residingViewId;
+
+    /**
+     * Used by media inflater.
+     */
     public AudioButton(Context context, AttributeSet attrs) {
```

Comment:
I took it out, because it looked like it wasn't being called, and the app crashed on login.